### PR TITLE
Support VolumeClaimGC in Workflow

### PR DESCRIPTION
**Pull request by szdr**
Support VolumeClaimGC
https://github.com/argoproj/argo-workflows/blob/master/sdks/python/client/docs/IoArgoprojWorkflowV1alpha1VolumeClaimGC.md

### CODE REVIEW

File: src/hera/volume_claim_gc.py
Comment (flaviuvadan) [inline src/hera/volume_claim_gc.py:24]: The [docs](https://argoproj.github.io/argo-workflows/fields/#volumeclaimgc) for volume claim GC suggest there's a single field at the moment, which is `strategy`. What do you think about using the `VolumeClaimGCStrategy` enum directly in a workflow `__init__` rather than wrapping it with a `VolumeClaimGC` `BaseModel`? I think `ImagePullPolicy` is similar to this - we use the enum directly. My decision boundary between using a base model and not using one is generally asking the question: "does this have more than 1 field and are there opportunities to construct logic on the fields?" If the answer is no, which I think is the case for image pull policy, GC, etc, then I suggest we don't use a `BaseModel`. Thoughts, @szdr?

Comment (szdr) [reply src/hera/volume_claim_gc.py:24]: Thanks for your comment, I agree with the idea of using VolumeClaimGCStrategy directly in __init__ of workflow.

File: src/hera/workflow.py
Comment (flaviuvadan) [inline src/hera/workflow.py:77]: I am not sure this is a good assumption 🤔  maybe some users may want to keep the volumes around post workflow completion. The same logic was applied when the other optional parameters were set to `None`. There's no way to know whether the user does want a TTL strategy, so Hera offers the option to add it but makes no assumptions about it, for now at least. Any chance we can set this to `None` as well? I think we are ready to merge right after that into 3.2.0 🚀

Comment (szdr) [reply src/hera/workflow.py:77]: You're right! I would leave the default value as None.

Comment (szdr) [reply src/hera/workflow.py]: https://github.com/argoproj-labs/hera-workflows/pull/147/commits/ed486defdddd891d13b69244ca3e445c49e05161

**Review by flaviuvadan — APPROVED** [review]
🚀

**Pull request merged**
